spec/ameba/rule/lint/duplicated_case_condition_spec.cr

```diff
@@ -0,0 +1,32 @@
+require "../../../spec_helper"
+
+module Ameba::Rule::Lint
+  describe DuplicatedCaseCondition do
+    subject = DuplicatedCaseCondition.new
+
+    it "passes if there are no duplicated `when` conditions" do
+      expect_no_issues subject, <<-CRYSTAL
+        case x
+        when .nil?
+          do_something
+        when Symbol
+          do_something_else
+        end
+        CRYSTAL
+    end
+
+    it "reports if there are a duplicated `when` conditions" do
+      source = expect_issue subject, <<-CRYSTAL
+        case x
+        when .nil?
```

Comment:
Does this work for this case:

```crystal
case x
when .foo?, .bar?
  do_something
when .bar?
  do_something_else
end
```

Reply:
Not at the moment.

Reply:
Good catch! I've added support for checking independent conditions.